fsspec/implementations/reference.py

```diff
@@ -126,11 +132,20 @@ def __init__(
             Encode urls as pandas.Categorical to reduce memory footprint if the ratio
             of the number of unique urls to total number of refs for each variable
             is greater than or equal to this number. (default 10)
+        engine: Literal["fastparquet","pyarrow"]
+            Engine choice for reading parquet files. (default is "fastparquet")
         """
+
+        from importlib.util import find_spec
+
+        if find_spec("pyarrow") is None:
```

Comment:
This gets called even if engine is "fastparquet", no? That would make this unusable if pyarrow isn't installed.

Reply:
(could test by monkeypatch on find_spec)

Reply:
Great catch!